app/Topology.hs

```diff
@@ -1,5 +1,7 @@
+{-# LANGUAGE NamedFieldPuns    #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE TypeApplications  #-}
```

Comment:
:roll_eyes:

Reply:
I do like this extension, it has a price (e.g. one cannot reorder type variable bindings when using `ExcplictForAll`) but it exposes a part of `GHC`'s `SysytemF`, which is convenient.